### ISSUE TITLE
If user change an idp claim, Pomerium don't refresh user session

### ISSUE THREAD

**Issue by xlejo** [pomerium/pomerium#2675]
## What happened?

When change the IDP claim of the user, like `groups` this not reflect in the route when Pomerium (with Redis) is running, showing  me a `403 Forbidden`, having to delete the Redis data and restart the Pomerium Databroker.

## What did you expect to happen?

If change the group of a user logged, Pomerium should refresh the user session.

## How'd it happen?

1. Make a route with `allowed_idp_claims` parameter, like `family_name` or in my case, `groups`.
2. Open a incognito tab, login and enter to the route.
3. Once you are logged in change the parameter `family_name` or `groups`, so that the user is not allowed to login. 
4. Reload the Pomerium configuration.
5. Close incognito tab and open again, login and enter to the route.
6. The user can logging without problems.

## What's your environment like?

- Kubernetes 1.22.2
- Pomerium version 0.15.3 (deployed with Pomerium Helm v24.0.0)
- Keycloak version 15.0.2

## What's your config.yaml?

```config.yaml
address: :443
grpc_address: :443
certificate_file: "/pomerium/tls/tls.crt"
certificate_key_file: "/pomerium/tls/tls.key"
certificate_authority_file: "/pomerium/ca/ca.crt"
http_redirect_addr: :80
authenticate_service_url: https://authenticate.example.tld
authorize_service_url: https://pomerium-authorize.pomerium.svc.cluster.local
databroker_service_url: https://pomerium-databroker.pomerium.svc.cluster.local
idp_provider: oidc
idp_scopes: [openid profile email]
idp_provider_url: https://keycloak.example.tld/auth/realms/example-realm
forward_auth_url: https://pomerium-proxy.pomerium
cookie_secret: <hidden>
shared_secret: <hidden>
idp_client_id: <hidden>
idp_client_secret: <hidden>
databroker_storage_type: redis
databroker_storage_connection_string: rediss://pomerium-redis-master.pomerium.svc.cluster.local
databroker_storage_tls_skip_verify: false 
policy:
  - from: https://app.example.tld
    to: http://app.example.internal.tld:8080
    pass_identity_headers: true
    allowed_idp_claims:
      groups:
      - /infraestructure
```

## Additional info

The idp claim `groups` is a mapper in Keycloak.

**Comment by boskisam** [comment]
I struggle with the same issue. Would like to see this resolved.
Every group change must be followed with databroker restart. (i have redis turned off).
Even when forcing logout client-side, changing groups, logging back in authenticate.example.com/.pomerium/ displays that the new groups are visible, but need to restart databroker for policy to refresh user groups.

**Comment by travisgroth** [comment]
Hi @boskisam sorry for the silence.  We anticipate releasing a patch that addresses this soon.

**Comment by travisgroth** [comment]
This is now fixed in [v0.15.6](https://github.com/pomerium/pomerium/releases/tag/v0.15.6).  Thank you for opening the issue @xlejo.
